From 3aef808eee43c973ae1fbf6e8769d89b7f0d355b Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Tue, 10 Jun 2014 14:47:42 +0000
Subject: [PATCH] deal with TH

Export modules referenced by it.

Should not need these icons in git-annex, so not worth using the Evil
Splicer.
---
 Network/Wai/Application/Static.hs | 4 ----
 WaiAppStatic/Storage/Embedded.hs  | 8 ++++----
 wai-app-static.cabal              | 4 +---
 3 files changed, 5 insertions(+), 11 deletions(-)

diff --git a/Network/Wai/Application/Static.hs b/Network/Wai/Application/Static.hs
index db2b835..b2c1aec 100644
--- a/Network/Wai/Application/Static.hs
+++ b/Network/Wai/Application/Static.hs
@@ -33,8 +33,6 @@ import Control.Monad.IO.Class (liftIO)
 
 import Blaze.ByteString.Builder (toByteString)
 
-import Data.FileEmbed (embedFile)
-
 import Data.Text (Text)
 import qualified Data.Text as T
 
@@ -198,8 +196,6 @@ staticAppPieces _ _ req sendResponse
         H.status405
         [("Content-Type", "text/plain")]
         "Only GET is supported"
-staticAppPieces _ [".hidden", "folder.png"] _ sendResponse = sendResponse $ W.responseLBS H.status200 [("Content-Type", "image/png")] $ L.fromChunks [$(embedFile "images/folder.png")]
-staticAppPieces _ [".hidden", "haskell.png"] _ sendResponse = sendResponse $ W.responseLBS H.status200 [("Content-Type", "image/png")] $ L.fromChunks [$(embedFile "images/haskell.png")]
 staticAppPieces ss rawPieces req sendResponse = liftIO $ do
     case toPieces rawPieces of
         Just pieces -> checkPieces ss pieces req >>= response
diff --git a/WaiAppStatic/Storage/Embedded.hs b/WaiAppStatic/Storage/Embedded.hs
index daa6e50..9873d4e 100644
--- a/WaiAppStatic/Storage/Embedded.hs
+++ b/WaiAppStatic/Storage/Embedded.hs
@@ -3,10 +3,10 @@ module WaiAppStatic.Storage.Embedded(
       embeddedSettings
 
     -- * Template Haskell
-    , Etag
-    , EmbeddableEntry(..)
-    , mkSettings
+    --, Etag
+    --, EmbeddableEntry(..)
+    --, mkSettings
     ) where
 
 import WaiAppStatic.Storage.Embedded.Runtime
-import WaiAppStatic.Storage.Embedded.TH
+--import WaiAppStatic.Storage.Embedded.TH
diff --git a/wai-app-static.cabal b/wai-app-static.cabal
index ef6f898..9a59d71 100644
--- a/wai-app-static.cabal
+++ b/wai-app-static.cabal
@@ -33,7 +33,6 @@ library
                    , containers                >= 0.2
                    , time                      >= 1.1.4
                    , old-locale                >= 1.0.0.2
-                   , file-embed                >= 0.0.3.1
                    , text                      >= 0.7
                    , blaze-builder             >= 0.2.1.4
                    , base64-bytestring         >= 0.1
@@ -61,9 +60,8 @@ library
                      WaiAppStatic.Listing
                      WaiAppStatic.Types
                      WaiAppStatic.CmdLine
-    other-modules:   Util
                      WaiAppStatic.Storage.Embedded.Runtime
-                     WaiAppStatic.Storage.Embedded.TH
+    other-modules:   Util
     ghc-options:     -Wall
     extensions:     CPP
 
-- 
2.0.0

